src/services/combineResponses.ts

```diff
@@ -72,11 +72,12 @@ export function mergeFrames(dest: DataFrame, source: DataFrame) {
   for (let i = 0; i < sourceTimeValues.length; i++) {
     const destTimeValues = destTimeField.values.slice(0) ?? [];
     const destNanosValues = destTimeField.nanos?.slice(0);
+    const destIdValues = destIdField?.values.slice(0) ?? [];
     const destIdx = resolveIdx(destTimeField, sourceTimeField, i);
 
-    const entryExistsInDest = compareTimestamps(
+    const entryExistsInDest = compareEntries(
       { ...destTimeField, values: destTimeValues, nanos: destNanosValues },
-      destIdField,
+      destIdField ? { ...destIdField, values: destIdValues } : destIdField,
```

Comment:
Not sure I understand, is this to create a copy of the `Field`?

Maybe still worth doing:
```suggestion
      destIdField ? { ...destIdField, values: destIdValues } : undefined,
```

Reply:
Great question. I've been using it so that the destination fields were not affected by incoming values, but you made me realize that now that's a leftover of the original implementation. I'll see if it still makes sense to keep it.